common/reviews/api/rush-azure-storage-build-cache-plugin.api.md

```diff
@@ -30,7 +33,9 @@ export abstract class AzureAuthenticationBase {
     deleteCachedCredentialsAsync(terminal: ITerminal): Promise<void>;
     protected abstract _getCacheIdParts(): string[];
     // (undocumented)
-    protected abstract _getCredentialFromDeviceCodeAsync(terminal: ITerminal, deviceCodeCredential: DeviceCodeCredential): Promise<ICredentialResult>;
+    protected abstract _getCredentialFromTokenAsync(terminal: ITerminal, tokenCredential: TokenCredential): Promise<ICredentialResult>;
```

Comment:
Can we avoid making this a breaking API change?

Reply:
Should I:
1. Rename `_getCredentialFromTokenAsync` back to `_getCredentialFromTokenAsync` still keeping the `DeviceCodeCredential ` to `TokenCredential` change to make the typing loose (`DeviceCodeCredential` extends `TokenCredential`)
2. Rename back to original with original typings and add a new`_getCredentialFromInteractiveBrowserAsync` with `InteractiveBrowserCredential` param?

Reply:
I'd go with option 3.

Reply:
Updated with a change that marks `_getCredentialFromDeviceCodeAsync` as deprecated.
I had to provide a default implementation for `_getCredentialFromDeviceCodeAsync` and `_getCredentialFromTokenAsync` since only one of them is required to be implemented.

I tried to enforce the typing with TypeScript interface merging but interface properties default to `public` and cannot be marked as `protected` so that would have been a breaking change.

The downside here is `throw new Error('_getCredentialFromDeviceCodeAsync not implemented')`.

Let me know if there's a better way to deal with this.